test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,314 @@
+	}
+	customDnsConfigMap, err := data.CreateConfigMap(configMap)
+	require.NoError(t, err, "failed to create custom DNS ConfigMap")
+
+	createCustomDnsPod(t, data, configMap.Name)
+
+	// set the custom DNS server IP address in Antrea configMap.
+	setDnsServerAddressInAntrea(t, data, dnsServiceIP)
+	defer setDnsServerAddressInAntrea(t, data, "") //reset after the test.
+
+	createFqdnPolicyInNamespace(t, data, testFQDN, "test-anp-fqdn", "custom-dns", "fqdn-cache-test")
+	createToolboxPod(t, data, dnsServiceIP)
+	curlTarget := testFQDN
+	curlFQDN := func() (string, error) {
+		cmd := []string{"curl", curlTarget}
+		stdout, stderr, err := data.RunCommandFromPod(data.testNamespace, toolboxPodName, toolboxContainerName, cmd)
+		if err != nil {
+			return "", fmt.Errorf("error when running command '%s' on Pod '%s': %v, stdout: <%v>, stderr: <%v>",
+				strings.Join(cmd, " "), toolboxPodName, err, stdout, stderr)
+		}
+		return stdout, nil
+	}
+
+	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+		stdout, err := curlFQDN()
+		assert.NoError(t, err)
+		t.Logf("Response of curl to FQDN: %s", stdout)
+	}, 2*time.Second, 100*time.Millisecond, "trying to curl the FQDN: ", testFQDN)
+
+	// confirm that the FQDN resolves to the expected IP address and store it to simulate caching of this IP associated with the FQDN.
+	t.Logf("Resolving FQDN to simulate caching the current IP inside Toolbox Pod...")
+	resolvedIP, err := data.runDNSQuery(toolboxPodName, toolboxContainerName, data.testNamespace, testFQDN, false, dnsServiceIP)
+	fqdnIP := resolvedIP.String()
+	require.NoError(t, err, "failed to resolve FQDN to an IP from toolbox Pod")
+	require.Equalf(t, agnHostPodOneIP, fqdnIP, "The IP set against the FQDN in the DNS server should be the same, but got %s instead of %s", fqdnIP, agnHostPodOneIP)
+	t.Logf("Successfully received the expected IP %s using the dig command against the FQDN", fqdnIP)
+
+	// update the IP address mapped to the FQDN in the custom DNS ConfigMap.
+	t.Logf("Updating host mapping in DNS server config to use new IP: %s", agnHostPodTwoIP)
+	customDnsConfigMap.Data = createDnsConfig(t, map[string]string{agnHostPodTwoIP: testFQDN}, dnsTTL)
+	require.NoError(t, data.UpdateConfigMap(customDnsConfigMap), "failed to update configmap with new IP")
+	t.Logf("Successfully updated DNS ConfigMap with new IP: %s", agnHostPodTwoIP)
+
+	// try to trigger an immediate refresh of the configmap by setting annotations in custom DNS server Pod, this way
+	// we try to bypass the kubelet sync period which may be as long as (1 minute by default) + TTL of ConfigMaps.
+	// Ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically
+	require.NoError(t, data.setPodAnnotation(data.testNamespace, "custom-dns-server", "test.antrea.io/random-value",
+		randSeq(8)), "failed to update custom DNS Pod annotation.")
+
+	curlTarget = fqdnIP
```

Comment:
nit: make `curlTarget` an argument to `curlFQDN`. Otherwise this looks out of place

Reply:
Yes , thats a nice suggestion.